File: Javascript/Algorithms/1374.js
/**
 * @param {number} n
 * @return {string}
 */
let generateTheString = n => {
    if (n == 1){
        return "a";
    }
    const odd = n & 1;
    let s = "a".repeat(odd? n - 2: n - 1);
    if (odd){
        s += "bc";
    } else {
        s += "b";
    }
    return s;
};

File: Python/Algorithms/1374.py
class Solution:
    
    def yield_chars(self, n):
        yield from repeat('a', n - 2) if n & 1 else repeat('a', n-1)
        yield "bc" if n & 1 else "b"
    
    def generateTheString(self, n: int) -> str:
        if n == 1:
            return 'a'
        return "".join(self.yield_chars(n)) 
